shell/client/accounts/login-buttons.js

```diff
@@ -65,8 +65,19 @@ function getActiveIdentityId(grains) {
 
 Template.accountButtons.helpers({
   displayName: function () {
+    if (Meteor.user() && !Meteor.user().loginIdentities) {
```

Comment:
I guess we could try to check the whether the main `identityProfile` subscription is ready? Seems somewhat tricky due to the [reactivity logic](https://github.com/sandstorm-io/sandstorm/blob/v0.173/shell/client/shell-client.js#L34-L53).

I've updated this so that it won't immediately return "loading..." in the case that the user is an identity user. Note, however, that when we display the login-with-identity interstitial, the login button is hidden.